src/prefect/flows.py

```diff
@@ -1887,10 +1883,8 @@ async def load_flow_from_flow_run(
         run_logger.debug(
             f"Importing flow code from module path {deployment.entrypoint}"
         )
-        flow = await run_sync_in_worker_thread(
```

Comment:
I just remembered that I added some logic to `run_sync_in_worker_thread` to handle `@sync_compatible` functions at the module level (e.g. `Block.load`). We will likely need a new solution for that with this `run_sync_in_worker_thread` call removed.

Reply:
I see, good test!! I'll poke at this and see if I can simplify the current code and if not, I'll just close this PR

Reply:
Going to close for now, the refactors I've tried feel weird.